network-config-analyzer/MultiLayerPropertiesSet.py

```diff
@@ -36,72 +37,293 @@
 """
 
 
+class UnlimitedHttpAttributes:
```

Comment:
Found an issue with `or` operator in some example as below (the last assertion fails).
Also, consider adding unit tests for this class separately as well.

    def test_basic_1(self):
        allow_all = UnlimitedHttpAttributes(True)
        allow_nothing = UnlimitedHttpAttributes()
        a = UnlimitedHttpAttributes()
        self.assertTrue(not a)
        print(a)
        a.values = {'a'}
        b = UnlimitedHttpAttributes()
        b.values = {'a'}
        b.negation = True
        print(a)
        print(b)
        print(a | b)
        print(a & b)
        self.assertTrue(a & b == allow_nothing)
        self.assertTrue(a | b == allow_all)

Reply:
good catch, thanks. will fix, do some more clean up and enhance testing of this class